app/models/concerns/orchestration/tftp.rb

```diff
@@ -51,37 +50,47 @@ def generate_pxe_template
 
     # work around for ensuring that people can use @host as well, as tftp templates were usually confusing.
     @host = self.host
+
     if build?
-      template = host.provisioning_template({:kind => host.operatingsystem.template_kind})
-      failure_missing_template unless template
-      template_name = template.name
+      begin
+        template = host.provisioning_template({:kind => kind})
+        return nil unless template
```

Comment:
Yes, I will be incorporating all your comments from now on. Will only comment if there is a problem.